src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/DeclarationHandler.java

```diff
@@ -325,4 +327,27 @@ TranslationUnitDeclaration handleTranslationUnit(CPPASTTranslationUnit translati
 
     return node;
   }
+
+  /**
+   * Returns a raw type string (that can be parsed by the {@link TypeParser} out of a cpp declarator
+   * and associated declaration specifiers.
+   *
+   * @param declarator the declarator
+   * @param declSpecifier the declaration specifier
+   * @return the type string
+   */
+  static String getTypeStringFromDeclarator(
+      IASTDeclarator declarator, IASTDeclSpecifier declSpecifier) {
+    // use the declaration specifier as basis
+    StringBuilder typeString = new StringBuilder(declSpecifier.getRawSignature());
+
+    // append names, pointer operators and array modifiers and such
+    for (IASTNode node : declarator.getChildren()) {
+      if (node instanceof IASTPointerOperator || node instanceof IASTArrayModifier) {
+        typeString.append(node.getRawSignature());
+      }
+    }
+
+    return typeString.toString();
+  }
```

Comment:
This seems to be less-error-prone than using `ctx.getRawSignature` for most cases. Nonetheless, we have a problem with function pointers, as by using this method to determine the raw type string instead of using `ctx.getRawSignature`, the fptr `void ((*single_param)(int))` is transformed into `void`. This is not the intended behaviour, since fptrs are addressed by their own type.

Reply:
That is why this is not used for function pointers (yet)

Basically function pointers are handled (for some reason) here:

https://github.com/Fraunhofer-AISEC/cpg/blob/8ff74381b5a2a753727da2eb4db890f79024c84a/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/DeclarationListHandler.java#L67-L68

I can add a comment to make it more clear

Reply:
Yes I have seen that if and that should be correct. However I tested the following snippet and I 
reached this code here: [https://github.com/Fraunhofer-AISEC/cpg/blob/8ff74381b5a2a753727da2eb4db890f79024c84a/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/DeclarationHandler.java#L107](https://github.com/Fraunhofer-AISEC/cpg/blob/8ff74381b5a2a753727da2eb4db890f79024c84a/src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/DeclarationHandler.java#L107)
```
int main(){
void ((*single_param)(int));
}
```

Reply:
In that case apparently the code is treated as a VariableDeclaration and therefore the if statement is true.

Reply:
Can you add this snippet somewhere to the existing function pointer unit tests. You can deliberately make the unit test fail and then I can have a look.

Reply:
> Can you add this snippet somewhere to the existing function pointer unit tests including an assert on how the type should be parsed. You can deliberately make the unit test fail and then I can have a look.

Ok I looked through the PR, and will await until you took that look.

Reply:
Yes I will add the snippet to a test